### PR TITLE
Meg ad extension for AMP

### PR DESCRIPTION
**Pull request by alunyov**
Please review this pull-request to support Meg as an ad extension for AMP.

### CODE REVIEW

**Comment by lannka** [comment]
we recently added new API: [renderStart](https://github.com/ampproject/amphtml/blob/master/ads/README.md#methods-available-to-the-ad)
Could you take a look and implement the APIs (renderStart + noContentAvailable)?
It helps AMP to provide a better ad loading experience.
Let me know if you have any questions regarding that.

**Comment by lannka** [comment]
And please also sign CLA if you haven't yet.

**Comment by alunyov** [comment]
@googlebot @lannka I signed CLA.

File: ads/meg.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+export function meg(global, data) {
+  validateData(data, ['code']);
+  const code = data.code;
+  const lang = global.encodeURIComponent(global.navigator.language);
+  const ref = global.encodeURIComponent(global.referrer);
+  const params = ['lang=' + lang, 'ref=' + ref].join('&');
+  const url = 'https://apps.meg.com/embedjs/' + code + '?' + params;
+  loadScript(global, url, () => {
+    global.context.renderStart();
```
Comment (alunyov) [inline ads/meg.js]: I've updated the pull request. Now `renderStart` and `noContentAvailable` have been wrapped into `onSuccess` and `onError` callbacks. Additionally, I call `noContentAvailable` if we are unable to load the main javascript bundle. I've added optional `onError` callback to the `loadScript` function. Unfortunately, I don't know what is the best way to test it.

File: ads/meg.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (lannka) [inline ads/meg.js]: 2016

File: ads/meg.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<!---
+Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (lannka) [inline ads/meg.md]: 2016

File: ads/meg.js (hunk 1 above)
Comment (alunyov) [inline ads/meg.js]: I've combined all commits into one and added tests and fixed the years in the headers.

**Pull request merged**